### PR TITLE
Stylus sidebar restructure

### PR DESCRIPTION
**Pull request by TucksonDev**
This PR attempts to restructure the sidebar menu for the Stylus section. It includes the following:
- Reorder the menu entries for a shorter sidebar and meaningful subsections to increase discoverability
- Adds the "Caching strategy" page to the "Gas and Ink" subsection, and adapts the Overview to mention it
- Modifies the "Reduce the size of WASM binaries" to remove outdated information and adapt the language to the new "CLI tools" subsection

### CODE REVIEW

File: website/sidebars.js [hunk 1]
```diff
@@ -437,30 +437,34 @@ const sidebars = {
...
+            {
+              type: 'link',
+              label: 'Rust crate docs',
+              href: 'https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html',
+            },
+          ],
+        },
+        {
+          type: 'category',
+          label: 'Gas, ink and caching',
```
Comment (mahsamoosavi) [inline website/sidebars.js:467]: Should we replace ink with Ink for the sake of consistency?

**Review by mahsamoosavi — CHANGES_REQUESTED** [review]
A tiny comment, LGTM otherwise!

File: website/sidebars.js (hunk 1 above)
Comment (TucksonDev) [reply website/sidebars.js:467]: I see that I haven't used that in other places though. I'll update them.

Comment (TucksonDev) [reply website/sidebars.js:467]: Updated, thanks for the flag!

**Review by anegg0 — APPROVED** [review]
Looking great to me. The content will evolve within the next two weeks, but I find this new structure much more straightforward.

File: website/sidebars.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (anegg0) [reply website/sidebars.js:467]: Are you OK with keeping ink low-cased, @mahsamoosavi?
I think it's better low-cased.
Also, the concept of `ink` is on its way to being deprecated.

[the remark this answers is not in the document]
Comment (mahsamoosavi) [reply website/sidebars.js:467]: Yes let's keep it low-cased.

**Review by mahsamoosavi — APPROVED** [review]

**Pull request merged**